src/sage/matroids/oriented_matroids/covector_oriented_matroid.py

```diff
@@ -0,0 +1,206 @@
+r"""
+Oriented matroid with covector axioms
+
+This implements an oriented matroid using the covector axioms.
+
+AUTHORS:
+
+- Aram Dermenjian (2019-07-12): Initial version
+"""
+
+##############################################################################
+#       Copyright (C) 2019 Aram Dermenjian <aram.dermenjian.math at gmail.com>
+#
+#  Distributed under the terms of the GNU General Public License (GPL)
+#
+#  The full text of the GPL is available at:
+#
+#                  http://www.gnu.org/licenses/
+##############################################################################
+
+from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+from sage.categories.sets_cat import Sets
+
+
+class CovectorOrientedMatroid(OrientedMatroid):
+    r"""
+    An oriented matroid implemented using covector axioms.
+
+    According to  Definition 3.7.1 in [BLSWZ1999]_ a *covector* of an oriented
+    matroid is any composition of cocircuits.
+
+    This implements an oriented matroid using the covector axioms. For this
+    let `\mathcal{L}` be a set of covectors and `E` a groundset. Then
+    a pair `M = (E,\mathcal{L})` is an oriented matroid using the covectors
+    axioms if (see Theorem 4.1.1 in [BLSWZ1999]_):
+
+    - `0 \in \mathcal{L}`
+    - `X \in \mathcal{L}` implies `-X \in \mathcal{L}`
+    - For all `X,Y \in \mathcal{L}`, `X \circ Y \in \mathcal{L}`
+    - For all `X,Y \in \mathcal{L}` and `e \in S(X,Y)` there exists a
+      `Z \in \mathcal{L}` such that `Z(e) = 0` and
+      `Z(f) = (X \circ Y)(f) = (Y \circ X)(f)` for all `f \notin S(X,Y)`.
+
+    INPUT:
+
+    - ``data`` -- a tuple containing :class:`SignedSubsetElement` elements or
+      data that can be used to construct :class:`SignedSubsetElement` elements
+    - ``groundset`` -- (default: ``None``) the groundset for the data; if not
+      provided, we grab the data from the signed subsets
+
+    EXAMPLES::
+
+        sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+        sage: M = OrientedMatroid([[1], [-1], [0]], groundset=['e'], key='covector')
+        sage: M
+        Covector oriented matroid of rank 1
+        sage: M.groundset()
+        ('e',)
+
+        sage: C = [[1,1,1], [1,1,0], [1,1,-1], [1,0,-1], [1,-1,-1], [0,-1,-1],
+        ....:      [-1,-1,-1], [0,1,1], [-1,1,1], [-1,0,1], [-1,-1,1], [-1,-1,0], [0,0,0]]
+        sage: M = OrientedMatroid(C, key='covector'); M
+        Covector oriented matroid of rank 2
+        sage: M.groundset()
+        (0, 1, 2)
+        sage: gs = ['h1', 'h2', 'h3']
+        sage: M = OrientedMatroid(C, key='covector', groundset=gs)
+        sage: M.groundset()
+        ('h1', 'h2', 'h3')
+
+    .. SEEALSO::
+
+        - :class:`~oriented_matroids.oriented_matroid.OrientedMatroid`
+    """
+
+    def __init__(self, data, groundset=None, category=None):
```

Comment:
```suggestion
    .. SEEALSO::

        - :class:`~sage.oriented_matroids.oriented_matroid.OrientedMatroid`
    """
    def __init__(self, data, groundset=None, category=None):
```
plus `__init__` needs a doctest.

Reply:
This seems pointless? Is there a reason that `__init__` would need a doctest considering the class itself already has doctests to test for the initialisation? Just seems like overkill, especially since it won't show up in the docs?

Or I guess we can make just `TEST` doctests? Is that what you mean for adding tests to `__init__`?

Reply:
It needs a doctest by our rules, and this is a good place to put a `TestSuite(M).run()` test.

Reply:
Sounds good =D Adding the `TestSuite` also helped catch a couple bugs! So good suggestion. (I didn't add a way to `==` and `hash` oriented matroids)